nilearn/surface/surface.py

```diff
@@ -968,25 +965,110 @@ def _gifti_img_to_mesh(gifti_img):
     return coords, faces
 
 
+def check_mesh_is_fsaverage(mesh):
+    """Check that :term:`mesh` data is either a :obj:`str`, or a :obj:`dict`
+    with sufficient entries. Basically ensures that the mesh data is
+    Freesurfer-like fsaverage data.
+
+    Used by plotting.surf_plotting.plot_img_on_surf and
+    plotting.html_surface._full_brain_info.
+    """
+    if isinstance(mesh, str):
+        # avoid circular imports
+        from nilearn.datasets import fetch_surf_fsaverage
+
+        return fetch_surf_fsaverage(mesh)
+    if not isinstance(mesh, Mapping):
+        raise TypeError(
+            "The mesh should be a str or a dictionary, "
+            f"you provided: {type(mesh).__name__}."
+        )
+    missing = {
+        "pial_left",
+        "pial_right",
+        "sulc_left",
+        "sulc_right",
+        "infl_left",
+        "infl_right",
+    }.difference(mesh.keys())
+    if missing:
+        raise ValueError(
+            f"{missing} {'are' if len(missing) > 1 else 'is'} "
+            "missing from the provided mesh dictionary"
+        )
+    return mesh
+
+
+def check_mesh_and_data(mesh, data):
```

Comment:
some of those checks look very much like some of the ones we do in PolyMesh, PolyData or SurfaceImage: maybe worth refactoring later...

Reply:
Yes, these functions are mostly used in the plotting module (which was using the old surface API and didn't have those checks otherwise)